src/main/scala/cromwell/binding/values/WdlArray.scala

```diff
@@ -10,5 +10,13 @@ case class WdlArray(wdlType: WdlArrayType, value: Seq[WdlValue]) extends WdlValu
   if (typesUsedInValue.size > 1) {
     throw new UnsupportedOperationException(s"Cannot construct array with a mixed types: $value")
   }
-  override def toWdlString(): String = s"[${value.map{x=>x.toWdlString}.mkString(", ")}]"
+
+  override def toWdlString(): String = s"[${value.map { x => x.toWdlString }.mkString(", ")}]"
+
+  def map[R <: WdlValue](f: WdlValue => R): WdlArray = {
+    value.map{f} match {
+      case s:Seq[R] if s.nonEmpty => WdlArray(WdlArrayType(s.head.wdlType), s)
+      case _ => this
```

Comment:
Yeah. 

Although now that I think about it you might be able to kinda do what @kshakir suggested via something like `Seq(head, _*)` which now that I type it out I think I like that, I thought it would look ugly.